src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -324,37 +345,42 @@ def _get_group_calc_docs(self, data: Data) -> list[CalculatedDataDocumentItem]:
         calculated_documents = []
         for group_block in data.block_list.group_blocks:
             for group_sample_data in group_block.group_data.sample_data:
+                for aggregated_entry in group_sample_data.aggregated_entries:
```

Comment:
could you please extract this new code to a helper? Triple-nested for loops are hard to follow.

Also, it seems like `data_sources` could be constructed outside this new loop since it uses `group_sample_data` but not `aggregated_entry`.